internal/jimm/upgrade/upgrade.go

```diff
@@ -23,26 +29,126 @@ type BootstrapManager interface {
 	StartBootstrapJob(ctx context.Context, user *openfga.User, params bootstrap.BootstrapParams) (string, error)
 }
 
+// JujuManager defines the juju manager methods required by the upgrade manager.
+type JujuManager interface {
```

Comment:
@SimoneDutto I think you were suggesting that we don't put this upgrade manager inside the Juju manager. But if we're using this logic from the Juju manager, I'm inclined to say it should just go in there. 
Wdyt? I'm not sure what we gain from making a separate manager here except for more layers of abstraction.

Reply:
We discussed it, all good.